core-s3/src/test/scala/io/aiven/guardian/kafka/s3/S3Spec.scala

```diff
@@ -99,26 +104,32 @@ trait S3Spec
             bucketsToCleanup.add(bucket)
     } yield ()
 
-  private def cleanBucket(bucket: String): Future[Unit] = (for {
-    check <- S3.checkIfBucketExists(bucket)
-    _ <- check match {
-           case BucketAccess.AccessDenied =>
-             Future {
-               logger.warn(
-                 s"Cannot delete bucket: $bucket due to having access denied. Please look into this as it can fill up your AWS account"
-               )
-             }
-           case BucketAccess.AccessGranted =>
-             logger.info(s"Cleaning up bucket: $bucket")
-             S3TestUtils.cleanAndDeleteBucket(bucket)
-           case BucketAccess.NotExists =>
-             Future {
-               logger.info(s"Not deleting bucket: $bucket since it no longer exists")
-             }
-         }
-
-  } yield ()).recover { case util.control.NonFatal(error) =>
-    logger.error(s"Error deleting bucket: $bucket", error)
+  private def cleanBucket(bucket: String): Future[Unit] = {
+    // The default logger takes an implicit TestData but
+    // having an implicit TestData here doesn't name sense
+    // since this function is run after all tests finish
+    val _logger = Logger(LoggerFactory.getLogger(getClass.getName))
```

Comment:
I can not come up with a more elegant solution right now, so I think we are good. But the fact we need to do this trick tells me we would need to design this harness better.